pkg/columns/columninfo.go

```diff
@@ -0,0 +1,221 @@
+	Tags         []string              // Tags can be used to dynamically include or exclude columns
+
+	fieldIndex    int          // used for the main struct
+	subFieldIndex []int        // used for embedded structs
+	kind          reflect.Kind // cached kind info from reflection
+	columnType    reflect.Type // cached type info from reflection
+}
+
+func (ci *Column[T]) fromTag(tag string) error {
+	tagInfo := strings.Split(tag, ",")
+	ci.Name = tagInfo[0]
+
+	tagInfo = tagInfo[1:]
+	for _, subTag := range tagInfo {
+		params := strings.SplitN(subTag, ":", 2)
+		paramsLen := len(params)
+		switch params[0] {
+		case "align":
+			if paramsLen == 1 {
+				return fmt.Errorf("missing alignment value for field %q", ci.Name)
+			}
+			switch params[1] {
+			case "left":
+				ci.Alignment = AlignLeft
+			case "right":
+				ci.Alignment = AlignRight
+			default:
+				return fmt.Errorf("invalid alignment %q for field %q", params[1], ci.Name)
+			}
+		case "ellipsis":
+			if paramsLen == 1 {
+				ci.EllipsisType = ellipsis.End
+				continue
+			}
+			switch params[1] {
+			case "end", "":
+				ci.EllipsisType = ellipsis.End
+			case "middle":
+				ci.EllipsisType = ellipsis.Middle
+			case "none":
+				ci.EllipsisType = ellipsis.None
+			case "start":
+				ci.EllipsisType = ellipsis.Start
+			default:
+				return fmt.Errorf("invalid ellipsis value %q for field %q", params[1], ci.Name)
+			}
+		case "fixed":
+			if paramsLen != 1 {
+				return fmt.Errorf("parameter fixed on field %q must not have a value", ci.Name)
+			}
+			ci.FixedWidth = true
+		case "group":
+			if paramsLen == 1 {
+				return fmt.Errorf("missing group value for field %q", ci.Name)
+			}
+			switch params[1] {
+			case "sum":
+				if !ci.columnType.ConvertibleTo(reflect.TypeOf(int(0))) {
+					return fmt.Errorf("cannot use sum on field %q of kind %q", ci.Name, ci.kind.String())
+				}
+				ci.GroupType = GroupTypeSum
+			default:
+				return fmt.Errorf("invalid sum value %q for field %q", params[1], ci.Name)
+			}
+		case "hide":
+			if paramsLen != 1 {
+				return fmt.Errorf("parameter hide on field %q must not have a value", ci.Name)
+			}
+			ci.Visible = false
+		case "order":
+			if paramsLen == 1 {
+				return fmt.Errorf("missing width value for field %q", ci.Name)
+			}
+			w, err := strconv.Atoi(params[1])
+			if err != nil {
+				return fmt.Errorf("invalid order value %q for field %q: %w", params[1], ci.Name, err)
+			}
+			ci.Order = w
+		case "precision":
+			if ci.kind != reflect.Float32 && ci.kind != reflect.Float64 {
+				return fmt.Errorf("field %q is not a float field and thereby cannot have precision defined", ci.Name)
+			}
+			if paramsLen == 1 {
+				return fmt.Errorf("missing precision value for field %q", ci.Name)
+			}
+			w, err := strconv.Atoi(params[1])
+			if err != nil {
+				return fmt.Errorf("invalid precision value %q for field %q: %w", params[1], ci.Name, err)
+			}
+			if w < 0 {
```

Comment:
Yeah, though actually it's a typo :) I planned to use `< -1`, since strconv.FormatFloat() - which we are using - allows the special precision of `-1` to format a float as precise as possible.